### ISSUE TITLE
Fix the MSO resources read issue during the state file refresh

### ISSUE THREAD

**Issue by sajagana** [CiscoDevNet/terraform-provider-mso#201]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### MSO version

* V x.x.x

### APIC version and APIC Platform for Site Level Resources

* V x.x.x and on-prem/cloud-aws/cloud-azure/all.

### Affected Resource(s)

MSO resource failed to update the resource id to nil when the object is manually removed from the NDO/MSO machine.

* mso_XXXXX

### Terraform Configuration Files

```hcl
# Copy-paste your Terraform configurations here - for large Terraform configs,
# please use a service like Dropbox and share a link to the ZIP file. For
# security, you can also encrypt the files using our GPG public key: https://keybase.io/hashicorp
```

### Expected Behavior

If the object is not present in the NDO/MSO then the object data should be removed from the state file during the `terraform apply`

### Actual Behavior
Currently, it does not remove the manually deleted object data from the terraform state file.

### Steps to Reproduce

1. `terraform apply`
2.  Delete the configured object from NDO/MSO manually via GUI
3.  `terraform apply`

### References

* #0000

**Comment by danischm** [comment]
I think there is already an issue open: https://github.com/CiscoDevNet/terraform-provider-mso/issues/183 .

**Comment by lhercot** [comment]
Lets keep this open as this is for other resources fixes as we only fixed the schema one in #183
